[Updating - Index Page] Adding Learning Objectives for New Chapters about Reactive Messaging and GraphQL

ttelang opened the pull request:
## Overview
This PR updates the learning objectives to cover new chapters included in the MicroProfile 7.1.

## Key Changes
- **Event-Driven and Reactive Microservices**: Added MicroProfile Reactive Messaging and integration with messaging systems (Kafka, etc.) for asynchronous, decoupled communication. 
- **GraphQL API Design**: Introduced MicroProfile GraphQL as a modern alternative to REST, enabling flexible, client-driven API development

### Expanded Existing Topics
- **MicroProfile APIs**: Explicitly added OpenAPI and Telemetry to the core API list (previously bundled under "other APIs")
- **Resilience Patterns**: Detailed specific fault tolerance patterns including retries, timeouts, circuit breakers, and bulkhead
- **Health Checks**: Distinguished between readiness and liveness checks for cloud environments
- **Observability**: Enhanced monitoring section to cover distributed tracing concepts and MicroProfile Telemetry alongside OpenTracing

ttelang committed:
Revise project overview and microservices descriptions

Updated the project overview and detailed sections on the MicroProfile e-Commerce application, including descriptions of key microservices and their functionalities.

ttelang committed:
Refine tutorial content on MicroProfile e-Commerce

Revised tutorial content for clarity and detail, enhancing descriptions of microservices and learning objectives.

ttelang committed:
Update content for clarity and accuracy in index.adoc

Emily-Jiang commented on modules/ROOT/pages/index.adoc:
```diff
@@ -92,44 +86,35 @@ we assume you have a basic understanding of RESTful Web Services.
-
-** Learn to trace microservice interactions with OpenTracing for enhanced observability.
-
+** Learn distributed tracing concepts and trace microservice interactions using OpenTracing and MicroProfile Telemetry for enhanced observability.
```
You should not mention OpenTracing as it is deprecated. Just mention MicroProfile Telemetry.

ttelang replied:
Removed OpenTracing

ttelang committed:
Update index.adoc

Removing OpenTracing from Learning Objectives as this is deprecated.

ttelang committed:
Update chapter02-00.adoc

Removing the reference to MicroProfile Starter as the applications is currently down. Will add the details back once the start.microprofile.io is hosted back.

ttelang committed:
Update chapter02-00.adoc

minor formating fix.

ttelang committed:
Update chapter02-02.adoc

Updating version numbers for updated components.

Emily-Jiang commented on modules/ROOT/pages/chapter02/chapter02-02.adoc:
```diff
@@ -141,25 +144,13 @@ The list below is provided to help you select the appropriate modules for your M
-    </dependency>
-  </dependencies>
-</project>
+<!-- MicroProfile Telemetry 2.1 API -->
```
There is no Telemetry API

ttelang replied:
@Emily-Jiang 

The Maven coordinates for Telemetry 2.1 API is documented in the official spec page in this link here:
[MicroProfile Telemetry 2.1 API](https://microprofile.io/specifications/telemetry/2-1/)

ttelang replied:
I have added note about the POM type requirement for the `microprofile-telemetry-api` artifact.

ttelang committed:
Add note on microprofile-telemetry-api artifact

Added note about the microprofile-telemetry-api artifact and its POM type requirement.

Emily-Jiang commented on modules/ROOT/pages/chapter02/chapter02-02.adoc:
```diff
@@ -141,27 +144,18 @@ The list below is provided to help you select the appropriate modules for your M
+<dependency>
+    <groupId>org.eclipse.microprofile.telemetry</groupId>
+    <artifactId>microprofile-telemetry-api</artifactId>
+    <version>2.1</version>
```
The lines won't work as there is no api jar for mp telemetry.

ttelang replied:
See details in the Note below.

NOTE: The `microprofile-telemetry-api` artifact is an aggregator POM that transitively pulls in `opentelemetry-api`, `opentelemetry-instrumentation-annotations`, and `opentelemetry-semconv`. The `<type>pom</type>` element is required so that Maven resolves it correctly rather than looking for a JAR.
